closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -474,12 +576,302 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+     * - Stop if a general symlink is loaded and we then find a
+     *   specific version that is higher.
+     * - Stop immediately if a specific version is loaded, as
+     *   anything after that will be a lower version.
+     */
+    static const char * const libNames[] = {
+#if defined(_AIX)
+        "libcrypto.a(libcrypto64.so)",      /* general symlink library name from archive file */
+        "libcrypto64.so",                   /* general symlink library name */
+        "libcrypto.a(libcrypto.so)",        /* general symlink library name from archive file */
+        "libcrypto.so",                     /* general symlink library name */
+        "libcrypto.a(libcrypto64.so.3)",    /* 3.x library name from archive file */
+        "libcrypto64.so.3",                 /* 3.x library name */
+        "libcrypto.a(libcrypto.so.3)",      /* 3.x library name from archive file */
+        "libcrypto.so.3",                   /* 3.x library name */
+        "libcrypto.a(libcrypto64.so.1.1)",  /* 1.1.x library name from archive file */
+        "libcrypto.so.1.1",                 /* 1.1.x library name */
+        "libcrypto.so.1.0.0",               /* 1.0.x library name */
+#elif defined(__APPLE__) /* defined(_AIX) */
+        "libcrypto.3.dylib",                /* 3.x library name */
+        "libcrypto.1.1.dylib",              /* 1.1.x library name */
+        "libcrypto.1.0.0.dylib",            /* 1.0.x library name */
+#elif defined(_WIN32) /* defined(__APPLE__) */
+        "libcrypto-3-x64.dll",              /* 3.x library name */
+        "libcrypto-1_1-x64.dll",            /* 1.1.x library name */
+        "libeay32.dll",                     /* old library name */
+#else /* defined(_WIN32) */
+        "libcrypto.so",                     /* general symlink library name */
+        "libcrypto.so.3",                   /* 3.x library name */
+        "libcrypto.so.1.1",                 /* 1.1.x library name */
+        "libcrypto.so.1.0.0",               /* 1.0.x library name */
+        "libcrypto.so.10",                  /* old library name */
+#endif /* defined(_AIX) */
+    };
+
+    const size_t numOfLibs = sizeof(libNames) / sizeof(libNames[0]);
+#if defined(_AIX)
+    const size_t num_of_generic = 4;
+#elif defined(__linux__) /* defined(_AIX) */
+    const size_t num_of_generic = 1;
+#else /* defined(__linux__) */
+    const size_t num_of_generic = 0;
+#endif /* defined(_AIX) */
+
+    void *result = NULL;
+    void *prevResult = NULL;
+    size_t i = 0;
+    long tempVersion = 0;
+    long previousVersion = 0;
+
+    /* If JAVA_HOME is not null or empty and no library has been loaded yet, try there. */
+    if ((NULL != chomepath) && ('\0' != *chomepath) && (NULL == crypto_library)) {
+#if defined(_WIN32)
+        static const char pathSuffix[] = "\\bin\\";
+#else /* defined(_WIN32) */
+        static const char pathSuffix[] = "/lib/";
+#endif /* defined(_WIN32) */
+        size_t path_len = strlen(chomepath) + sizeof(pathSuffix) - 1;
```

Comment:
A blank line before lines 683 and 685 would be welcome (and consistent with lines 660 and 669).

Reply:
Blank lines added.